dashboard/app/models/script.rb

```diff
@@ -336,10 +336,14 @@ def starting_level
     candidate_level
   end
 
-  # Find the lockable or non-locakble stage based on its relative position.
-  # Raises `ActiveRecord::RecordNotFound` if no matching stage is found.
-  def stage_by_relative_position(position, lockable = false)
-    lessons.where(lockable: lockable).find_by!(relative_position: position)
+  # Find the lesson based on its relative position, lockable value, and if it has a lesson plan.
+  # Raises `ActiveRecord::RecordNotFound` if no matching lesson is found.
+  def lesson_by_relative_position(position, survey = false)
+    if survey
```

Comment:
would you be willing to call this `use_survey_url` instead? I still think `/survey/` is a good choice for the url, but I am worried that it is a bit confusing to use this term like this in the code without explaining it, given that "survey" already has a definition in our system, i.e. an anonymous level group.